src/librustc_mir/borrow_check.rs

```diff
@@ -549,7 +564,7 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
                     _ => unreachable!(),
                 };
                 self.access_lvalue(
-                    context, (lvalue, span), (Shallow(Some(af)), Read(ReadKind::Copy)), flow_state);
+                    context, (lvalue, span), (Shallow(Some(af)), Read(ReadKind::Copy)), LocalMutationIsAllowed::No, flow_state);
```

Comment:
`dropandreplace` is an assignment of a type with a destructor, so this should be an error only if the value is already initialized.

e.g. the string example:
```Rust
let s = String::new();
s = String::new();
```